src/cryptography/hazmat/primitives/ciphers/aead.py

```diff
@@ -365,10 +365,21 @@ def _check_params(
         data: bytes,
         associated_data: typing.List[bytes],
     ) -> None:
-        utils._check_bytes("data", data)
+        utils._check_byteslike("data", data)
         if len(data) == 0:
             raise ValueError("data must not be zero length")
-        if not isinstance(associated_data, list) or not all(
-            isinstance(x, bytes) for x in associated_data
-        ):
-            raise TypeError("associated_data must be a list of bytes or None")
+
+        msg = "associated_data must be a list of bytes-like objects or None"
+        raise_except = False
+
+        if not isinstance(associated_data, list):
+            raise_except = True
+        else:
+            for x in associated_data:
+                try:
+                    utils._check_byteslike("associated_data", x)
+                except TypeError:
+                    raise_except = True
+                    break
+        if raise_except:
+            raise TypeError(msg)
```

Comment:
```suggestion
        if not isinstance(associated_data, list):
            raise TypeError("associated_data must be a list of bytes-like objects or None")
        for x in associated_data:
            utils._check_byteslike("associated_data elements", x)
```

Reply:
sure thing, I just fear the 80char per column flake check, we'll see

Reply:
Hmm, yes it may require some slightly line wrapping -- `black` or `ruff` may be able to auto-format it.